4 - AnimalShelter/Code/OperationResult.cs

```diff
@@ -0,0 +1,25 @@
+﻿using AnimalShelterCode;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AnimalShelterCode
+{
+    public class OperationResult
+    {
+        public OperationResult(Animal animal)
+        {
+            Animal = animal;
+            OperationSuccess = true;
+        }
+        public OperationResult(string error)
+        {
+            Error = error;
+            OperationSuccess = false;
+        }
+        internal Animal Animal { get; private set; }
+        internal string Error { get; private set; }
+        internal bool OperationSuccess { get; private set; }
```

Comment:
What you could do is, instead of setting this property, the property could return a value on the spot that evaluates the animal property. For example:

`public bool OperationSuccess { get { return Animal != null && Error == null; } }`